presto-main/src/main/java/com/facebook/presto/cost/CostComparator.java

```diff
@@ -59,6 +60,11 @@ public int compare(Session session, PlanNodeCostEstimate left, PlanNodeCostEstim
         requireNonNull(right, "right is null");
         checkArgument(!left.hasUnknownComponents() && !right.hasUnknownComponents(), "cannot compare unknown costs");
 
+        long queryMaxMemory = getQueryMaxMemory(session).toBytes();
```

Comment:
It would be nice to use dedicated session property here, which could be set to query max memory by default. This is because that memory cost is not exactly the memory consumption.